### ISSUE TITLE
Simplify NuGet package usage from msbuild

### ISSUE THREAD

**Issue by iskiselev** [lextm/obfuscar#124]
It is hard to integrate obfuscar from NuGet package to MsBuild scripts, as it hard to resolve actual path to obfuscar.console.exe. It could be greatly simplify, if package will contain file build/obfuscar.props with text like:
```xml
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <obfuscar>$(MSBuildThisFileDirectory)..\tools\obfuscar.console.exe</obfuscar>
  </PropertyGroup>
</Project>
```
With it call to obfuscar from MsBuild file will be possible with `$(obfuscar)` ms-build expression.

**Comment by lextm** [comment]
My personal belief is that obfuscation should happen outside of MSBuild, so I demonstrated that in the basin example repo,

https://github.com/lextm/obfuscar_example/blob/master/test.bat

By using package reference and the NuGet cache folder, it is not quite difficult to locate the console tool.

But, does your `$(obfuscar)` expression work for both `packages.config` and package references?

**Comment by iskiselev** [comment]
`$(obfuscar)` expression will work with any NuGet reference type - packages.config, project.json, PackageReference. When NuGet installs package with props/target file in packages.config based project, it will add conditional import of props file to the beginning of project and targets to the end with additional target for verification that packages were restored. You could look how it works by playing with IlRepack NuGet package, that use this trick.
Calculation of folder where tool was installed is really tricky - folder could be modified by NuGet.config on different levels, and packages.config and other references use different folder, and you still need save in some additional place currently installed version.

**Comment by iskiselev** [comment]
Using of obfuscation outside msbuild script is also not always good solution - obfuscated binaries could be input artifacts for other projects - for example, creation of installer.

**Comment by lextm** [comment]
Implemented in 2.2.11 release.

**Comment by lextm** [comment]
#134 gives an opportunity to install the tool globally. That should gradually be the primary way to acquire Obfuscar binaries in the future.
